src/lib.rs

```diff
@@ -49,7 +49,7 @@ pub enum Value {
 
     Symbol(intern::Symbol),
     ForeignFn(ForeignFunction<()>),
-    Lambda(Procedure),
+    Lambda(Procedure, bool),
```

Comment:
boolean blindness strikes again! It's "do I get my args evaluated or not?". The design decision I'm least happy with.